lib/diagnostic.rb

```diff
@@ -5,36 +5,40 @@ class Diagnostic
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.max
   end
+  # Jason Liang's Comment:
+  # I wanted to write enum.max.length but each entry
+  # implements a length method.
+
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(:+)
```

Comment:
Correct, but only in Ruby v2.0 and above. In older versions of Ruby, you need to explicitly add the `&` before the symbol:
`enum.reduce(&:+)`